src/main/java/edu/ucdavis/dss/ipa/entities/SupportAssignment.java

```diff
@@ -97,4 +98,14 @@ public long getSectionGroupIdentification() {
             return 0;
         }
     }
+
+    @Transient
+    @JsonIgnore
+    public boolean isOpenToReview() {
+        String termCode = this.getSectionGroup().getTermCode();
+        String term = termCode.substring(termCode.length() - 2, termCode.length());
+        List<String> openTerms = this.getSectionGroup().getCourse().getSchedule().getInstructorSupportCallReviewAsTerms();
```

Comment:
The many layers makes me uncomfortable here. Not sure how best to clean it up though.

Reply:
Hmmm.
The information is needed, so it will need to be obtained either through ORM entity chaining or a service layer backed by SQL query.

I could see this moving isOpenToReview to a service rather than an entity helper, as its touching quite a few other entities while doing its work. I usually think of entity helpers for simple calculations involving only itself.